src/figuresClasses.ts

```diff
@@ -1,19 +1,191 @@
 export interface Figure {
-
+  shape: 'triangle' | 'circle' | 'rectangle';
+  color: 'red' | 'green' | 'blue';
+  getArea(): number;
 }
 
 export class Triangle {
+  shape: 'triangle';
+
+  color: 'red' | 'green' | 'blue';
+
+  a: number;
+
+  b: number;
+
+  c: number;
+
+  constructor(
+    color: 'red' | 'green' | 'blue',
+    a: number,
+    b: number,
+    c: number,
+  ) {
+    if (a >= (b + c) || b >= (a + c) || c >= (a + b)) {
+      throw new Error('invalid side length for triangle');
+    }
 
+    this.shape = 'triangle';
+    this.color = color;
+    this.a = a;
+    this.b = b;
+    this.c = c;
+  }
+
+  getArea(): number {
+    const p = (this.a + this.b + this.c) / 2;
+
+    return Math.floor(Math.sqrt(p * (p - this.a)
+      * (p - this.b) * (p - this.c)) * 100) / 100;
```

Comment:
Looks better in my opinion, but it's up to you 
```suggestion
    return (
      Math.floor(
        Math.sqrt(p * (p - this.a) * (p - this.b) * (p - this.c)) * 100
      ) / 100;
    )
```